### PR TITLE
Adding text objects management to tests

### PR DESCRIPTION
**Pull request by geonexus**
### REVIEWERS

@pratid @flopezag @jframos 
### DESCRIPTION
- Adding CRD operations to manage text objects using swift.
- Adding tests to create texts objects into the Object Store.

### CODE REVIEW

File: fiware-region-sanity-tests/commons/fiware_cloud_test_case.py [hunk 1]
```diff
@@ -400,6 +400,8 @@ def reset_world_containers(cls, world, suite=False):
         # release resources to ensure a clean world
         for container in list(world['containers']):
             try:
+                for object in cls.swift_operations.get_container(container)[-1]:
```
Comment (jframos) [inline fiware-region-sanity-tests/commons/fiware_cloud_test_case.py:404]: The 'get_container' method can return more than one container? 
Why you try to remove objects only in the last container?

File: fiware-region-sanity-tests/commons/swift_operations.py [hunk 2]
```diff
@@ -85,13 +85,47 @@ def list_containers(self, name_prefix=None):
...
+    def create_text_object(self, container_name, local_file, object_name):
+        """
+        Creates a new text object into a container
+        :param container_name: Name of the container
+        :return: None if container was created and the error message if something failed.
+        """
+        with open(local_file, 'r') as test_file:
+            self.client.put_object(container_name, object_name,
+                                        contents=test_file.read(),
+                                        content_type=TEXT_PLAIN)
```
Comment (jframos) [inline fiware-region-sanity-tests/commons/swift_operations.py]: PEP8 compliant ?

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 3]
```diff
@@ -61,15 +62,55 @@ def test_create_get_and_delete_container(self):
...
+
+        ## Uploading the object
+
+        response = self.swift_operations.create_text_object(containerName, SWIFT_RESOURCES_PATH
+                                                            + TEST_TEXT_OBJECT_PREFIX + TEXT_EXTENSION,
+                                                            textObjectName + TEXT_EXTENSION)
+        origin = hashlib.md5(open(SWIFT_RESOURCES_PATH + TEST_TEXT_OBJECT_PREFIX + TEXT_EXTENSION, 'rb')
+                          .read()).hexdigest()
+
+        self.assertIsNone(response)
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: Add a message error for each assert, in order to see a comprehensible reason of problems in result reports.

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 4]
```diff
@@ -61,15 +62,55 @@ def test_create_get_and_delete_container(self):
             self.assertRaises(e)
             self.logger.debug("%s container was successfully removed from the object storage" % containerName)
 
-    def test_create_container(self):
+    def test_create_container_with_object_deleting_them(self):
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: I think we should have different test cases for each operation...

File: fiware-region-sanity-tests/commons/swift_operations.py [hunk 5]
```diff
@@ -85,13 +85,47 @@ def list_containers(self, name_prefix=None):
...
+
+    def get_text_object(self, container_name, object_name, download_path):
+        """
+        Downloads a text object from a container
+        :param container_name: Name of the container
+        :param object_name: Name of the container
```
Comment (flopezag) [inline fiware-region-sanity-tests/commons/swift_operations.py]: -        :param object_name: Name of the container?

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 6]
```diff
@@ -61,15 +62,55 @@ def test_create_get_and_delete_container(self):
...
         """
-        Test if it can be possible create a new container into the object storage.
+        Test if it can be possible create a new container into the object storage, upload a text file, download it and remove the
```
Comment (flopezag) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: pep8 compliant?

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py (hunk 4 above)
Comment (flopezag) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: +1

File: fiware-region-sanity-tests/commons/swift_operations.py [hunk 7]
```diff
@@ -85,13 +85,47 @@ def list_containers(self, name_prefix=None):
...
+        :param download_path: Path to download folder
+        :return: None if container was created and the error message if something failed.
+        """
+        obj = self.client.get_object(container_name, object_name)
```
Comment (flopezag) [inline fiware-region-sanity-tests/commons/swift_operations.py:117]: I do not follow the code, but it is not possible that self.client.get_object returns an error or null obj?

File: fiware-region-sanity-tests/commons/constants.py [hunk 8]
```diff
@@ -66,6 +66,11 @@
 TEST_NETWORK_PREFIX = "testing_network"
 TEST_ROUTER_PREFIX = "testing_router"
 TEST_CONTAINER_PREFIX = "testing_container"
+TEST_TEXT_OBJECT_PREFIX = "testing_text_object"
+TEXT_EXTENSION = ".txt"
```
Comment (pratid) [inline fiware-region-sanity-tests/commons/constants.py]: Maybe rename it to TEST_TEXT_FILE_EXTENSION?

File: fiware-region-sanity-tests/commons/constants.py [hunk 9]
```diff
@@ -83,3 +88,6 @@
 
 # ENDPOINTS
 ENDPOINT_TYPE_PUBLIC_URL = "publicURL"
+
+# TYPES
+TEXT_PLAIN = "text/plain"
```
Comment (pratid) [inline fiware-region-sanity-tests/commons/constants.py]: Maybe rename it to TEST_TEXT_CONTENT_TYPE?

File: fiware-region-sanity-tests/commons/fiware_cloud_test_case.py (hunk 1 above)
Comment (geonexus) [inline fiware-region-sanity-tests/commons/fiware_cloud_test_case.py:404]: Get container returns container info.
[-1] matches with the list of objects inside the container.

File: fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py [hunk 10]
```diff
@@ -51,25 +54,114 @@ def test_create_get_and_delete_container(self):
...
 
+    def test_delete_container(self):
+        """
+        Test if it can be possible delete a container.
+        """
+        suffix = datetime.utcnow().strftime('%Y%m%d%H%M%S')
+        containerName = TEST_CONTAINER_PREFIX + suffix
+
+        response = self.swift_operations.create_container(containerName)
+        self.assertIsNone(response)
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: Add a message error for each assert, in order to see a comprehensible reason of problems in result reports.

Comment (geonexus) [inline fiware-region-sanity-tests/tests/fiware_region_object_storage_tests.py]: fixed

**Comment by flopezag** [comment]
LGTM

**Comment by pratid** [comment]
LGTM

**Comment by jframos** [comment]
LGTM

**Pull request merged**